### PR TITLE
Improved test coverage

### PR DESCRIPTION
**Pull request by cygnusv**
## What this does: ##
- Includes the tests in the coverage report. There were some missing lines in the tests due to bugs in the definition of the tests. 
- Improves the coverage of the `umbral` module.
   - Test for reconstruction of activated capsule with consistent CFrags
   - Test for `Point.__sub__`
- Includes a check of `M` and `N` in `pre.split_rekey`.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -8,4 +8,4 @@ install:
     - pip install pipenv --upgrade
     - pipenv install --dev --three
 script:
-    - pipenv run py.test --cov=umbral
+    - pipenv run py.test --cov=umbral --cov=tests
```
Comment (KPrasch) [inline .travis.yml]: Let's remove this file since we've migrated to circleci.

File: tests/conftest.py [hunk 2]
```diff
@@ -22,6 +22,13 @@
     (50, 30)
 ]
 
+wrong_parameters = [
```
Comment (KPrasch) [inline tests/conftest.py]: Cool! We need to keep the number of parameterized tests to a minimum for test performance. What's a sensible place to test with these invalid values?

File: tests/test_primitives/test_point/test_point_serializers.py [hunk 3]
```diff
@@ -71,19 +81,15 @@ def test_invalid_points(random_ec_point2):
...
-        assert False
+    with pytest.raises(InternalError) as e:
+        _point = Point.from_bytes(point_bytes)
+
+    # We want to catch specific InternalExceptions:
+    # - Point not in the curve (code 107)
+    # - Invalid compressed point (code 110)
+    # https://github.com/openssl/openssl/blob/master/include/openssl/ecerr.h#L228
+    assert e.value.err_code[0].reason in (107, 110)
+
```
Comment (KPrasch) [inline tests/test_primitives/test_point/test_point_serializers.py]: Muuuch nicer. +1

File: tests/test_correctness.py [hunk 4]
```diff
@@ -228,3 +229,43 @@ def test_m_of_n(N, M, alices_keys, bobs_keys):
 
     sym_key_from_capsule = pre._open_capsule(capsule, priv_key_bob)
     assert sym_key == sym_key_from_capsule
+
+
+@pytest.mark.parametrize("N, M", [(N,M) for (N,M) in parameters if M>1])
+def test_inconsistent_cfrags(N, M, alices_keys, bobs_keys):
```
Comment (tuxxy) [inline tests/test_correctness.py]: 👍 This is great!
